lib/rspec-puppet-facts.rb

```diff
@@ -404,3 +408,11 @@ def self.facter_version_for_puppet_version(puppet_version)
     :default => RspecPuppetFacts.facter_version_for_puppet_version(Puppet.version)
   c.add_setting :facterdb_string_keys, :default => false
 end
+
+# monkey patch the Hash Class to add deep_merge
+class ::Hash
+  def deep_merge(second)
+    merger = proc { |_, v1, v2| Hash === v1 && Hash === v2 ? v1.merge(v2, &merger) : Array === v1 && Array === v2 ? v1 | v2 : [:undefined, nil, :nil].include?(v2) ? v1 : v2 }
+    merge(second.to_h, &merger)
+  end
+end
```

Comment:
Can we use https://rubygems.org/gems/deep_merge instead?

Reply:
I had seen that gem and thought about using it, but didn't want to introduce extra dependencies without your go ahead. 
I'll look to get it added